templates/submission/submissionMetadataFormTitleFields.tpl

```diff
@@ -20,5 +20,9 @@
 	{fbvElement label="common.subtitle.tip" type="text" multilingual=true name="subtitle" id="subtitle" value=$subtitle readonly=$readOnly}
 {/fbvFormSection}
 {fbvFormSection title="common.abstract" for="abstract" required=$abstractsRequired}
-	{fbvElement type="textarea" multilingual=true name="abstract" id="abstract" value=$abstract rich="extended" readonly=$readOnly}
+	{if $wordCount}
+		<input type="hidden" name="wordCount" value="{$wordCount|escape}" />
```

Comment:
What's the hidden field for?

Reply:
To provide the wordCount as a form parameter/field, so when the form has to be re-displayed, to have it -- then the readInputData is used, the variable got so it is known for re-display...

Reply:
How is this value used when re-displaying the form? Presumably, TinyMCE is re-calculating the word count from the abstract value. And on the server side, we should also be calculating the word count from the abstract. 

My concern is that we shouldn't be accepting any user-controlled input that specifies the word count. This value should always be calculated from the abstract text.

Reply:
This is only the number defined in the section settings i.e. the max. word limit. Currently the value is not used -- it is just there to know to enable the wordcount plugin for that tinymce. So theoretically it could be any number now, it is only important that the variable is set i.e. read by the form reload/re-display. 
But I've just thinking/wanted to also use it to mark the wordcount in the tinymce status bar with red color -- as you suggested -- when that number is reached.
The form is then valuated using the the abstract text.
So at the moment I believe it's ok to have it like this, right?

On the other side: today I was thinking that maybe it would be good to used the word count tinymce calculated -- so we would be 100% sure that the form validation considers the same number as it is displayed in the tinymce. But in that case I would need another hidden input fields -- for the form to know that tinymce word number. Now I will have to figure something else i.e. maybe still use the abstract test as you said and try to match the tinymce wordcount algorithm...

Reply:
> The form is then valuated using the the abstract text.
> So at the moment I believe it's ok to have it like this, right?

Instead of listening for the data to be sent with the form, we should check the abstract length setting from the section settings whenever the form is reloaded. For instance, maybe I'm a Journal Manager modifying an abstract. I hit the limit, open a new tab, change the word count to empty, then switch back to the tab. My form will still submit a word count, but that setting has changed. So best to go to the source each time.

> On the other side: today I was thinking that maybe it would be good to use the word count tinymce calculated

We could use that for client-side validation, before it's sent to the server. But we can't trust that number because the end-user could change it to whatever they want.

The docs look like they include the regex used to calculate word count: https://www.tinymce.com/docs/plugins/wordcount/ Maybe we can use that? (Although it says Removed in 4.5.0).

Ugh, the code looks pretty complicated: https://github.com/tinymce/tinymce/blob/master/src/plugins/wordcount/src/main/js/text/WordGetter.js

Hmm...

Reply:
OK, I'll take a look to then differently re-display the form.

Phew... Maybe then not to use the wordcount plugin but to implement it on our own, separating only by space? I just do now know where to display the number... Hmmm... :-P

Reply:
Yeah if it's more work than it's worth don't worry about it. You can get the content of tinymce directly from the input field, and then try out that regex in the wordcount documents on the frontend and backend to keep it synced.